src/clients/bridges/ZKSyncAdapter.ts

```diff
@@ -0,0 +1,315 @@
+      const l2Signer = this.getSigner(chainId);
+      // @dev Can re-use ABI from L1 weth as its the same for the purposes of this function.
+      const contract = new Contract(l2WethAddress, CONTRACT_ADDRESSES[1].weth.abi, l2Signer);
+      const method = "deposit";
+      const value = ethBalance.sub(threshold);
+      this.logger.debug({ at: this.getName(), message: "Wrapping ETH", threshold, value, ethBalance });
+      return (await txnClient.submit(chainId, [{ contract, chainId, method, args: [], value }]))[0];
+    }
+    return null;
+  }
+
+  async checkTokenApprovals(address: string, l1Tokens: string[]): Promise<void> {
+    const associatedL1Bridges = l1Tokens
+      // We unwrap WETH to send it over as ETH so it doesn't require an approval.
+      .filter((token) => !this.isWeth(token) && !this.supportedERC20s.includes(token))
+      .map(() => this.getL1ERC20BridgeContract().address);
+    await this.checkAndSendTokenApprovals(address, l1Tokens, associatedL1Bridges);
+  }
+
+  private getMailboxContract(): Contract {
+    const { hubChainId } = this;
+    const zkSyncMailboxContractData = CONTRACT_ADDRESSES[hubChainId]?.zkSyncMailbox;
+    if (!zkSyncMailboxContractData) {
+      throw new Error(`zkSyncMailboxContractData not found for chain ${hubChainId}`);
+    }
+    return new Contract(zkSyncMailboxContractData.address, zkSyncMailboxContractData.abi, this.getSigner(hubChainId));
+  }
+
+  private getL1ERC20BridgeContract(): Contract {
+    const { hubChainId } = this;
+    const l1Erc20BridgeContractData = CONTRACT_ADDRESSES[hubChainId]?.zkSyncDefaultErc20Bridge;
+    if (!l1Erc20BridgeContractData) {
+      throw new Error(`l1Erc20BridgeContractData not found for chain ${hubChainId}`);
+    }
+    return new Contract(l1Erc20BridgeContractData.address, l1Erc20BridgeContractData.abi, this.getSigner(hubChainId));
+  }
+
+  private async getL2ERC20BridgeContract(): Promise<Contract> {
+    const { provider } = this.spokePoolClients[this.chainId].spokePool;
+    const l2Erc20BridgeContractData = CONTRACT_ADDRESSES[this.chainId]?.zkSyncDefaultErc20Bridge;
+    if (!l2Erc20BridgeContractData) {
+      throw new Error(`l2Erc20BridgeContractData not found for chain ${this.chainId}`);
+    }
+    return new Contract(l2Erc20BridgeContractData.address, l2Erc20BridgeContractData.abi, provider);
+  }
+
+  /**
+   * @dev The L1Messenger contract is on zkSync and emits L1MessageSent when a mesage is sent back to mainnet.
+   */
+  private getl1Messenger(): Contract {
+    const { provider } = this.spokePoolClients[this.chainId].spokePool;
+    const { L1_MESSENGER, L1_MESSENGER_ADDRESS } = zksync.utils; // @todo Do we want to hardcode these?
+    return new Contract(L1_MESSENGER_ADDRESS, L1_MESSENGER, provider);
+  }
+
+  isSupportedToken(l1Token: string): l1Token is string {
+    const relevantSymbol = Object.values(TOKEN_SYMBOLS_MAP).find(({ addresses }) => {
+      return Object.values(addresses).includes(l1Token);
```

Comment:
This is not as safe as it could be. We know `l1Token` is an L1 token address so we shouldn't check for all addresses on all chains. There might be (very unlikely) an address collision where one token's L2 token address == another token's L1 address.
```suggestion
      return addresses[0] === l1Token;
```

Reply:
Tightened it to be more strict then the suggested edit - it will only test against the hub pool chain:
```
const relevantSymbol = Object.values(TOKEN_SYMBOLS_MAP).find(
      ({ addresses }) => addresses[this.hubChainId] === l1Token
    )?.symbol;
```

Reply:
https://github.com/across-protocol/relayer-v2/commit/d9475bc9fd9e8fe8717a50d254a3177150e33aef